### ISSUE TITLE
`SlevomatCodingStandard.Arrays.DisallowPartiallyKeyed`: False positive with anonymous functions and foreach loops

### ISSUE THREAD

**Issue by HeySora** [slevomat/coding-standard#1753]
PHP_CodeSniffer version 3.13.0
`slevomat/coding-standard` version 8.18.1

This example does not generate any error:

```php
[
    function () {
        foreach ($something as $value) {
            echo $value;
        }
    },
    function () {
        foreach ($something as $value) {
            echo $value;
        }
    },
];
```

This example does:

```php
[
    function () {
        foreach ($something as $key => $value) {
            echo $value;
        }
    },
    function () {
        foreach ($something as $value) {
            echo $value;
        }
    },
];
```

The reason being that `$keyValues` is populated with this:
```
[key:SlevomatCodingStandard\Helpers\ArrayKeyValue:private] => function () {
                foreach ($something as $key
```

the `=>` operator in the foreach loop is causing this false positive.
